sgx-isa/Cargo.toml

```diff
@@ -23,7 +23,10 @@ categories = ["hardware-support"]
 [dependencies]
 # External dependencies
 bitflags = "1" # MIT/Apache-2.0
+serde = { version = "1.0.84", features = ["derive"], optional = true } #MIT/Apache-2.0
+serde-big-array = { version = "0.1.5", optional = true } # MIT/Apache-2.0
 
 [features]
 large_array_derive = []
 sgxstd = []
+serde-support = ["serde", "serde-big-array"]
```

Comment:
I see. I think it's not worth using this dependency right now, since it's only used in one place and it prevents us from having the feature name we want. It will also make fixing the situation in the future (when const generics become a thing) harder.

How about we just add this in `lib.rs`?
```rust
#[cfg(feature = "serde")]
mod array_64 {
    use serde::{
        de::{Deserializer, Error, SeqAccess, Visitor},
        ser::{SerializeTuple, Serializer},
    };
    use core::fmt;

    const LEN: usize = 64;

    pub fn serialize<S: Serializer>(array: &[u8; LEN], serializer: S) -> Result<S::Ok, S::Error> {
        let mut seq = serializer.serialize_tuple(LEN)?;
        for elem in &array[..] {
            seq.serialize_element(elem)?;
        }
        seq.end()
    }

    pub fn deserialize<'de, D: Deserializer<'de>>(deserializer: D) -> Result<[u8; LEN], D::Error> {
        struct ArrayVisitor;
        impl<'de> Visitor<'de> for ArrayVisitor {
            type Value = [u8; LEN];
            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
                write!(formatter, "an array of length 64")
            }
            fn visit_seq<A: SeqAccess<'de>>(self, mut seq: A) -> Result<[u8; LEN], A::Error> {
                let mut arr = [0; LEN];
                for i in 0..LEN {
                    arr[i] = seq
                        .next_element()?
                        .ok_or_else(|| A::Error::invalid_length(i, &self))?;
                }
                Ok(arr)
            }
        }
        deserializer.deserialize_tuple(64, ArrayVisitor)
    }
}
```

Reply:
Could you give an example of how to use this?
`#[cfg_attr(feature = "serde", serde(default = "array_64::serialize"))]` is clearly not right...

Reply:
`Serialize` is fine, but trying to add `Deserialize` on `TargetInfo`, for example, leads to 
`the trait 'core::default::Default' is not implemented for '[u8; 456]'` once I remove `serde-big-array`.

Reply:
Same way you were using big array before, so `serde(with = "array_64")`

Reply:
I'm not sure what your point is about `TargetInfo`, that's already working correctly in the current PR?

Reply:
Ah, thanks, you're right. This helps. I think I can get it working now.